src/EditorFeatures/Core/StackTraceExplorer/StackTraceExplorerUtilities.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            // MemberAccessExpression is [Expression].[Identifier], and Identifier is the 
+            // method name.
+            var typeExpression = compilationUnit.MethodDeclaration.MemberAccessExpression.Left;
+            var fullyQualifiedTypeName = typeExpression.ToString();
+            var typeName = typeExpression is StackFrameQualifiedNameNode qualifiedName
+                ? qualifiedName.Right.ToString()
+                : typeExpression.ToString();
+
+            RoslynDebug.AssertNotNull(fullyQualifiedTypeName);
+
+            var methodIdentifier = compilationUnit.MethodDeclaration.MemberAccessExpression.Right;
+            var methodTypeArguments = compilationUnit.MethodDeclaration.TypeArguments;
+            var methodArguments = compilationUnit.MethodDeclaration.ArgumentList;
+
+            var methodName = methodIdentifier.ToString();
+
+            //
+            // Do a first pass to find projects with the type name to check first 
+            //
+            using var _ = PooledObjects.ArrayBuilder<Project>.GetInstance(out var candidateProjects);
+            foreach (var project in solution.Projects)
+            {
+                if (!project.SupportsCompilation)
+                {
+                    continue;
+                }
+
+                var containsSymbol = await project.ContainsSymbolsWithNameAsync(
+                    (name) => name == typeName,
+                    SymbolFilter.Type,
+                    cancellationToken).ConfigureAwait(false);
+
+                if (containsSymbol)
+                {
+                    var matchingMethods = await GetMatchingMembersFromCompilationAsync(project).ConfigureAwait(false);
+                    if (matchingMethods.Any())
+                    {
+                        return await GetDefinitionAsync(matchingMethods[0]).ConfigureAwait(false);
+                    }
+                }
+                else
+                {
+                    candidateProjects.Add(project);
+                }
+            }
+
+            //
+            // Do a second pass to check the remaining compilations
+            // for the symbol, which may be a metadata symbol in the compilation
+            //
+            foreach (var project in candidateProjects)
+            {
+                var matchingMethods = await GetMatchingMembersFromCompilationAsync(project).ConfigureAwait(false);
+                if (matchingMethods.Any())
+                {
+                    return await GetDefinitionAsync(matchingMethods[0]).ConfigureAwait(false);
+                }
+            }
+
+            return null;
+
+            //
+            // Local Functions
+            //
+
+            async Task<ImmutableArray<IMethodSymbol>> GetMatchingMembersFromCompilationAsync(Project project)
+            {
+                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
we already checked source above.  so really we just want to check metdata.  so we should have a hashset tracking which PE references we looked at.  we should then walk each project and walk the PE refs of each, seeing if it's the first time we're seeing it.  if so, we should check our metadata-index for that pe-ref to see if it's a hit.

Reply:
IMO we should do this in a follow up PR, it seems complex enough to warrant it's own commit and some feedback on the correct way to do it.